src/main/java/com/altigee/needoff/auth/model/UserPrincipal.java

```diff
@@ -0,0 +1,56 @@
+package com.altigee.needoff.auth.model;
+
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+public class UserPrincipal implements UserDetails {
```

Comment:
You can use Lombok's `@Getter`

Reply:
actually no need to introduce getters sine every method is explicitly defined